src/api/component.js

```diff
@@ -186,4 +179,12 @@ export default class Component extends HTMLElement {
 
     return Ctor;
   }
+
+  static shouldRender (elem, prev, curr) {
+    for (let name in prev) {
```

Comment:
That would mean something mutated `props` after an instance was created which I would think is something we shouldn't support.